include/aspect/simulator_signals.h

```diff
@@ -148,6 +146,33 @@ namespace aspect
     boost::signals2::signal<void (typename parallel::distributed::Triangulation<dim> &)>  post_refinement_load_user_data;
 
     /**
+    * A signal that is called before the creation of every checkpoint.
+    * This signal allows for registering functions that store data
+    * related to mesh cells.
+    *
+    * The functions that connect to this signal must take a reference
+    * to a parallel::distributed::Triangulation object as
+    * argument. This argument will point to the triangulation used by
+    * the Simulator class.
+    */
+    boost::signals2::signal<void (typename parallel::distributed::Triangulation<dim> &)>  pre_serialization_store_user_data;
```

Comment:
since serialization can mean both storing and loading, maybe rename this and the next signal as `pre_store_...` and `post_load_...`?